src/py_profet/interactive_plots/curve_utils.py

```diff
@@ -19,19 +21,23 @@ def get_color_bar(labels, stress_score_config):
 def filter_df(df, node_name=None, i_socket=None, i_mc=None, time_range=(), bw_range=(), lat_range=()):
     mask = [True] * len(df)
     if node_name is not None:
-        mask &= df['node_name'] == node_name
+        mask_node_name = df['node_name'] == node_name
+        mask = [a and b for a, b in zip(mask, mask_node_name)]
     if i_socket is not None:
-        mask &= df['socket'] == int(i_socket)
+        mask_socket = df['socket'] == int(i_socket)
+        mask = [a and b for a, b in zip(mask, mask_socket)]
```

Comment:
No veig quin era el problema en com estava fet anteriorment. Aquest filtratge amb el zip és força menys eficient i donaria el mateix resultat, no?

Reply:
Per algún motiu, sortía aquest error amb el codi antic i en mode debug:

PROFET-PARAVER/src/py_profet/interactive_plots/curve_utils.py:22: FutureWarning:

Logical ops (and, or, xor) between Pandas objects and dtype-less sequences (e.g. list, tuple) are deprecated and will raise in a future version. Wrap the object in a Series, Index, or np.array before operating instead.

I fent aquests canvis això desapareix.

Reply:
Ah, ja veig quin és el problema amb aquest warning. S'està queixant que la mask inicial és tipus llista:
`mask = [True] * len(df)`

Aquest warning se solucionarà si ho passes a Series:
`mask = pd.Series([True] * len(df))`

A part d'aquest aquest canvi, crec que seria millor deixar el codi com estava, és més eficient, simple i intuïtiu.